src/lib/AuthButton.svelte

```diff
@@ -0,0 +1,60 @@
+<script lang="ts">
+  import { onMount } from "svelte";
+  import { auth, pb } from "$lib/stores/authStore";
+  import { goto } from "$app/navigation";
+
+  let isAuthenticated = false;
+
+  onMount(auth.subscribe((value) => {
+    isAuthenticated = value.isAuthenticated;
+  }));
+
+  if (typeof document !== "undefined") {
+    pb.authStore.loadFromCookie(document.cookie);
+    pb.authStore.onChange(() => {
+      document.cookie = pb.authStore.exportToCookie({ httpOnly: false });
+    });
+  }
+
+  function goHome() {
+    if (window.location.pathname === '/') {
+      window.location.reload();
+    } else {
+      goto('/');
+    }
+  }
+
+  async function login() {
+    const authData = await pb.collection("users").authWithOAuth2({ provider: "google" });
+
+    const meta = authData.meta;
+
+    if (meta?.isNew) {
```

Comment:
```suggestion
    if (authData.meta?.isNew) {
```

Reply:
Enig med at ekstrahering av meta ikke er nødvendig.

Reply:
> Enig med at ekstrahering av meta ikke er nødvendig.

Jeg mener det gjør koden mye ryddigere, da den samme variabelen blir brukt to ganger til rett etter i koden.